Extend energy scan mode in NXenergydispersion

lukaspie opened the pull request:
Initial draft for discussion of energy scan mode of the energy dispersive element.
We need some better description of transmission and jittered/dithered energy scan modes.

Fixes #126

rettigl commented on contributed_definitions/NXenergydispersion.nxdl.xml:
```diff
@@ -69,19 +69,79 @@
+                     .. _12.66: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.66
+                </doc>
+            </item>
+            <item value="fixed_energy">
+                <doc>
+                     In the fixed energy (FE) mode, the intensity for one single kinetic energy is measured for a
+                     specified time. This mode is particulary useful during setup or alignment of the
+                     electron analyzer, for analysis of stability of the excitation source or for sample
+                     alignment.
+                     
+                     Since the mode measures intensity as a function of time, the difference in channel signals 
+                     is not of interest. Threfore, the signals from all channels are summed.
+                     
+                     Synonyms: FE mode
+                </doc>
```
If I understand this correctly, this does not produce an energy spectrum, and therefore is not so much of relevance for NXmpes. maybe we should restrict the enumeration there?

lukaspie replied:
For me, this would be pretty helpful to keep because this is the way I perform my alignment scans. These are typically saved in the same file as the actual data. Alignment scans are measurement points vs. intensity at one fixed kinetic energy.

rettigl replied:
sure, fine, but if we require energy axis these won't be compatible with NXmpes

lukaspie replied:
Can we not just have a 0D energy axis, i.e. simply write the fixed energy that was used for alignment?

I am hesitant to remove this since at some point, we will not have the possibility to write some portion of data that typically occurs in MPES-related experiments (with the machines our vendors produce) with our definition. Same as for the PEEM app def.

rettigl commented on contributed_definitions/NXenergydispersion.nxdl.xml:
```diff
@@ -69,19 +69,79 @@
+                     measurements, their energy resolution is relatively lower.
+                </doc>
+            </item>
+            <item value="transmission">
+                <doc>
+                     MISSING DOC.
+                     
+                         This concept is related to term `&lt;term&gt;`_ of the &lt;spec&gt; standard.
+                     .. _&lt;term&gt;: &lt;url&gt;
+                </doc>
+            </item>
```
I think this was intended to be the same as fixed_transmission. Probably can be removed.

lukaspie replied:
done

rettigl commented on contributed_definitions/NXenergydispersion.nxdl.xml:
```diff
@@ -69,19 +69,79 @@
+                     .. _&lt;term&gt;: &lt;url&gt;
+                </doc>
+            </item>
+            <item value="jittered">
```
"dither"

lukaspie replied:
fixed

rettigl commented on contributed_definitions/NXenergydispersion.nxdl.xml:
```diff
@@ -69,19 +69,79 @@
+            </item>
+            <item value="jittered">
+                <doc>
+                     MISSING DOC.
+                     
+                         This concept is related to term `&lt;term&gt;`_ of the &lt;spec&gt; standard.
+                     .. _&lt;term&gt;: &lt;url&gt;
```
In dither acquisition mode, the kinetic energy of the analyzer is randomly varied by a small value around a central value and at fixed pass energy. This allows reducing or removing inhomogeneities of the detector efficiency, such as e.g. imposed by a mesh in front of the detector. Mostly relevant for CCD/DLD type of detectors.

lukaspie replied:
Thanks, I put it into the docstring

rettigl reviewed (approved):
Here's a suggestion for the missing docstring. I am wonding if we should also add a mode for TOFs. Before we used "fixed", but this has been removed now, and "snapshot" does not really capture this case.

domna commented on contributed_definitions/NXenergydispersion.nxdl.xml:
```diff
@@ -69,19 +69,79 @@
-            <item value="sweep"/>
+            <item value="fixed_analyser_transmission">
+                <doc>
+                     constant ΔE mode, where the electron retardation (i.e., the fraction of pass energy to
+                     kinetic energy, R = (EK - WF/Ep)) is scanned, but the pass energy Ep is kept constant.
+                     Here, WF is the spectrometer work function.
+                     This mode is often used in XPS/UPS because the energy resolution does not change with
+                     changing energy (due to the constant pass energy).
+                     
+                     Synonyms: constant ΔE mode, constant analyser energy mode, CAE mode, FAT mode
+                     
+                         This concept is related to term `12.64`_ of the ISO 18115-1:2023 standard.
+                     .. _12.64: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.64
```
Something, I recently encountered while fixing the parsing of iso paragraphs was that enumeration items are rendered into a single line. Previously, it did even not build the documentation when there was a documentation with more than one paragraph. Now it just concatenates everything.

So we need to find a solution for this rather long docstrings here as this will be horrible to read I think. But probably this is something we need to fix in the html generation.

domna replied:
It's a bit cluttered but is fine for now I think.

<img width="1511" alt="Bildschirmfoto 2024-01-05 um 16 27 16" src="https://github.com/FAIRmat-NFDI/nexus_definitions/assets/1014422/dbe7af12-b2d2-4434-a9ea-6b05eae3ecb9">

lukaspie replied:
Ok, that is indeed a bit cluttered :D is that on nyaml or on the HTML generation? In non-xref cases, the HTML build does recognize paragraph breaks, right?

domna replied:
See #142

domna commented on contributed_definitions/nyaml/NXenergydispersion.yaml:
```diff
@@ -40,18 +40,78 @@ NXenergydispersion(NXobject):
+        - |
+          MISSING DOC.
+        - |
+          xref:
+            spec: <spec>
+            term: <term>
+            url: <url>
```
This produces a warning during documentation build and breaks the building of the process. Maybe without `<` and `>`?

domna replied:
Error:

```
Warning, treated as error:
/Users/domna/fairmat/nexus_definitions/build/manual/source/classes/contributed_definitions/NXenergydispersion.rst:112:Unknown target name: "<term> <<url>>".
make[1]: *** [html] Error 2
make: *** [local] Error 2
```

lukaspie replied:
removed

lukaspie commented:
- added  `AXISNAME_depends` to `NXdata`
- added `NXdata_mpes` with explicitly spelled axis names -> extends NXmpes, not in use yet (needs base class inheritance)
- removed name axes from NXmpes, keep energy axis -> still with `NXdata`, to be replaced by `NXdata_mpes`

lukaspie commented:
I think this can be merged unless there is any other open questions.

Regarding the mode for TOFs, I suggest we propose what we have here in the workshop and ask for some more input.

domna reviewed (approved):
Good to go from my side
